tlatools/org.lamport.tlatools/test/tlc2/value/ValueInputOutputStreamTest.java

```diff
@@ -115,4 +122,83 @@ public void testWriteNat() throws IOException {
 		assertEquals(0, in.readNat());
 		in.close();
 	}
+	
```

Comment:
I propose adding a TLA+-level test where the specification contains several definitions that make up the set of strings `S`. The spec's correctness property should involve string equality.  The input `readExternal` should deserialize strings that are present in `S` and those that are not.  

Background: Two `UniqueString`s instance are equal regardless of their `toString` representation. Instead, the position of the String in TLC's InternTable determines if two `UniqueString` instances are equal. The position in the `InternTable` is assigned non-deterministically, making it possible for two `UniqueString` instances to be equal (unequal) even if their string reps are equal (unequal).

Reply:
Yes, that seems worth it. Will add.

Reply:
(well, as much as I can without perhaps having access to IOUtils? I think the community modules are not available during test right now... but that doesn't mean I can't do something less direct)

Reply:
https://github.com/tlaplus/tlaplus/pull/1103